src/test/java/DecodeTest.java

```diff
@@ -0,0 +1,112 @@
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import decoder.Decoder;
+import encoder.Encoder;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Validate that decoding the protobuf file to json works properly. */
+@RunWith(JUnit4.class)
+public final class DecodeTest {
+  private static InputStream encoderInput;
+  private static OutputStream encoderOutput;
+  private static String decoderOutputFile;
+  private static InputStream decoderInput;
+  private static OutputStream decoderOutput;
+  private static String[] testInputArray;
+  private static String testInput;
+  private static String decoderInputFile;
+
+  @Before
+  public void setUp() throws SecurityException, FileNotFoundException {
+    decoderOutputFile = "decodeTestOutput.json";
+    decoderInputFile = "testProtoFile";
+    // Attempt to open both the input and output streams for testing.
+    testInputArray =
+        new String[] {
+          "gps_timestamp lat lng alt roll_deg pitch_deg yaw_deg",
+          "1140885738.407125 48.129872 11.582905 10.015034 -177.971817 -3.228588 10.700191",
+          "1140885738.475212 48.129872 11.582905 10.014367 -177.982559 -3.510518 10.743480"
+        };
+    testInput = testInputArray[0] + "\n" + testInputArray[1] + "\n" + testInputArray[2];
+    encoderInput = new ByteArrayInputStream(testInput.getBytes());
+    encoderOutput = new FileOutputStream(decoderInputFile);
+    decoderOutput = new FileOutputStream(decoderOutputFile);
```

Comment:
Prefer to make these local variables in each test. It's much easier to visually verify a test when you can see the input and expected output right next to each other. (This is of course balanced against efficiency, but I don't think that's an issue here.)

https://docs.google.com/document/d/1vLre-K3xneE4ROU6Hwkm4r0ujJ9_uFkG01Zu4zaoRzw/edit?usp=sharing